pkg/processor/util/clock/clock.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2017 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clock
```

Comment:
Why not implement this with `uint64` and throw away all the atomic code?

Reply:
> Is uint64 assignment atomic
Yes, reads/writes are 64 bit in width on 64 bit architectures

The goal is not to save code, but to prevent the use of locks

Reply:
How can I prevent a lock with `uint64`?

Reply:
clock writes the time every `resolution` seconds to a uint64, consumers read the time by reading that uint64. Since read/write to uint64 is naturally atomic you don't need to sync this by any means